addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationPlan.java

```diff
@@ -96,6 +97,8 @@ public AutomationPlan(ExtensionAutomation ext, File file) throws IOException {
                 if (job != null) {
                     try {
                         job = job.newJob();
+                        Map<?, ?> jobData = env.replaceMapVarsRecursive(rawJobData);
```

Comment:
The replacement needs to happen just before the execution, otherwise the plan loaded contains no variables (e.g. if the plan is saved it will no longer contain the vars; in the GUI it no longer shows them).